crates/fuel-core/src/state/historical_rocksdb.rs

```diff
@@ -94,6 +103,7 @@ where
         Ok(Self {
             state_rewind_policy,
             db,
+            modifications_history_migration_in_progress: true,
```

Comment:
It would be nice to add logic that checks do we have any entries in the V1. If not, set this to false

Reply:
Yes, the latest version of the PR does not have this check anymore, and instead we check directly if we are able to fetch at least one change for the v1 history.